gen-happy-sdist/gen-happy-sdist.cabal

```diff
@@ -0,0 +1,15 @@
+cabal-version:       1.12
+name:                gen-alex-sdist
+version:             0
+synopsis:            Tool for generating `alex` sdist
+category:            Development
+build-type:          Simple
+
+executable gen-alex-sdist
+  default-language:    Haskell2010
+  main-is:             Main.hs
+  build-depends:       base                    < 5
+                     , cpphs      >= 1.20.8 && < 1.21
```

Comment:
Is cpphs needed? Why?

Reply:
Templates are processed with CPP. `cpphs` is safe choice for CPP, than simply relying on whatever version one might have on their machine, i.e. we'll produce deterministic results. Also easy to use as a library.

Yet, I really don't know. `gen-alex-sdist` uses `cpphs` as well: https://github.com/simonmar/alex/blob/1e7af951077eb39ce3a0b669fbbb64ae4e3f62b2/gen-alex-sdist/gen-alex-sdist.cabal#L12

Reply:
indeed; Oleg's rationale is basically why I used cpphs so we'd have deterministic OS-independent results (and wouldn't have to deal with compat issues between various `cpp` implementations which have subtly different corner-case behaviour)

Reply:
OK